gen-models/src/operations.rs

```diff
@@ -344,17 +350,18 @@ impl FileAddition {
         conn: &Connection,
         file_path: &str,
         file_type: FileTypes,
+        checksum_override: Option<HashId>,
     ) -> Result<FileAddition, FileAdditionError> {
         let (absolute_file_path, relative_file_path) =
             FileAddition::normalize_file_paths(conn, file_path);
 
-        let checksum = if relative_file_path.is_empty() {
-            HashId::convert_str("empty")
+        let checksum = if let Some(checksum_override) = checksum_override {
+            checksum_override
         } else {
-            let checksum_path = if absolute_file_path.is_empty() {
```

Comment:
This prioritizes using the relative path as the basis for the checksum, hopefully that makes sense